GrandCentralBoard/Widgets/Bonus/Scene/Bubble.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  Created by Krzysztof Werys on 23/02/16.
+//  Copyright © 2016 Krzysztof Werys. All rights reserved.
+//
+
+import SpriteKit
+
+class Bubble: SKSpriteNode {
+    
+    private var bonus: Int = 0
+    private let initialSize = CGSize(width: 100, height: 100)
+    
+    init(person: Person) {
+        guard let image = person.bubbleImage.remoteImage ?? person.bubbleImage.localImage ?? UIImage(named: "placeholder") else { fatalError()}
+        
+        self.bonus = person.bonus
+        let texture = SKTexture(image: image.cropToCircle())
+        super.init(texture: texture, color: UIColor.clearColor(), size: initialSize)
+        
+        setUpPhysicsBody(texture, size: initialSize, person: person)
+        self.name = person.name
+        
+        let scaleBy: CGFloat = 1 + CGFloat(person.bonus) / 100
+        setScale(scaleBy)
+    }
+    
+    private func findImage(person: Person) -> UIImage? {
+        return UIImage(named: "placeholder")
+    }
+    
+    private func setUpPhysicsBody(texture: SKTexture, size: CGSize, person: Person) {
+        let spacing: CGFloat = 1
+        physicsBody = SKPhysicsBody(circleOfRadius: initialSize.width / 2 + spacing)
+        physicsBody?.restitution = 0.0
+        physicsBody?.friction = 0.3
+        physicsBody?.linearDamping = 0.5
+        physicsBody?.allowsRotation = false
+    }
+    
+    required init(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func updateImage(image: UIImage) {
+        let newImage = image.cropToCircle()
+        self.texture = SKTexture(image: newImage)
+    }
+    
+    func updateWithNewBonus(newBonus: Int) {
+        let difference = newBonus - self.bonus
+        guard difference > 0 else { return }
```

Comment:
We increase bonus and run animation only if the value of bonus changes for a bigger one.
I'll add a comment.